lightning/src/chain/channelmonitor.rs

```diff
@@ -1022,7 +1029,7 @@ impl<ChanSigner: ChannelKeys> ChannelMonitor<ChanSigner> {
 	/// counterparty commitment transaction's secret, they are de facto pruned (we can use revocation key).
 	fn provide_secret(&mut self, idx: u64, secret: [u8; 32]) -> Result<(), MonitorUpdateError> {
 		if let Err(()) = self.commitment_secrets.provide_secret(idx, secret) {
-			return Err(MonitorUpdateError("Previous secret did not match new one"));
+			return Err(MonitorUpdateError::NoPersistMonitor("Previous secret did not match new one"));
```

Comment:
I think you can ride of this as CounterpartyCommitmentSecrets::provide_secret doesn't write if it return an error. Thus we can store the ChannelMonitor cleanly there is no risk of gross state.

More interestingly, a provide_secret() failure here shouldn't happen as this check is already done inside Channel in `revoke_and_ack()`. Note, I think we should keep it but that would be a fair hint of a buggy or compromised coordinator and we should Take Action On This (tm). This is unrelated to this PR, so will add yet-another note in #604 for myself.